### PR TITLE
[Doc] Add a release guide

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
... and an empty ChangeLog file.

This patch partially is based on #453 (it includes one extra place that should be updated for release)

Preview: https://github.com/lukaszstolarczuk/unified-memory-framework/blob/release-guide/RELEASE_STEPS.md

### CODE REVIEW

File: RELEASE_STEPS.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+- Add an entry to ChangeLog, remember to change the day of the week in the release date
+  - For major releases mention API and ABI compatibility with the previous release
+- Update project's version in a few places:
+  - set the new $VERSION in `project` function in the top-level `CMakeLists.txt`
+  - `release` variable in `scripts/docs_config/conf.py` (for docs)
+  - `UMF_VERSION` variable in `.github/workflows/basic.yml` (for installation test)
+- For major releases update ABI version in `.map` and `.def` files
+  - these files are defined for all public libraries (`libumf` and `proxy_lib`, at the moment)
+- Commit these changes and tag the release:
+  - git commit -a -S -m "$VERSION release"
```
Comment (szadam) [inline RELEASE_STEPS.md:30]: You could add bash format here and in other places with git commands. I think it will make it clearer to read:
```bash
$ git commit -a -S -m "$VERSION release"
```

**Review by szadam — CHANGES_REQUESTED** [review]

File: RELEASE_STEPS.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  - git tag -a -s -m "Version $VERSION" $VERSION
+- For major/minor release:
+  - If stable branch for this release is not created and it's required, create it:
+    - git checkout -b v$VER.x (e.g., `v0.1.x`)
```
Comment (kswiecicki) [inline RELEASE_STEPS.md:34]: This command is exactly the same as the one at the beginning of `To do a release: ...`. Perhaps the info about creating this stable branch should be included there as well.

File: RELEASE_STEPS.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  - set the new $VERSION in `project` function in the top-level `CMakeLists.txt`
+  - `release` variable in `scripts/docs_config/conf.py` (for docs)
+  - `UMF_VERSION` variable in `.github/workflows/basic.yml` (for installation test)
+- For major releases update ABI version in `.map` and `.def` files
+  - these files are defined for all public libraries (`libumf` and `proxy_lib`, at the moment)
+- Commit these changes and tag the release:
+  - git commit -a -S -m "$VERSION release"
+  - git tag -a -s -m "Version $VERSION" $VERSION
+- For major/minor release:
+  - If stable branch for this release is not created and it's required, create it:
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply RELEASE_STEPS.md]: I don't have strong opinion about it, it could be either way I think... so perhaps we leave it to decide to the "release doer"...?

File: RELEASE_STEPS.md (hunk 2 above)
Comment (lukaszstolarczuk) [reply RELEASE_STEPS.md]: I re-did it a little, adding more info - pls let me know if it's ok now.

File: RELEASE_STEPS.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  set $VER = new major+minor only version (e.g., 0.1)
+```
+
+**Note:**
+> Before doing the final release, it's recommended to prepare a pre-release version - a "release candidate"
+> (or "rc" in short). This requires adding, e.g., `-rc1` to the VERSION string. When all tests and checks
+> end properly, you can follow up with the final release. If any fix is required, it should be included in
+> another rc version (e.g., `-rc2`).
+
+At the moment, UMF releases are aligned with oneAPI releases. Once all changes planned for
```
Comment (PatKamin) [inline RELEASE_STEPS.md:17]: Why the alignment information is here? I think the user wouldn't know what to do with this information. Does it mean that there will be UMF versions named after oneAPI releases or sth else?

Comment (lukaszstolarczuk) [reply RELEASE_STEPS.md:17]: tbh, I followed the UR writing in here:

https://github.com/oneapi-src/unified-runtime?tab=readme-ov-file#release-process

In my understanding it means only that we will release mostly for oneAPI usage and we'll align the date for the oneAPI release date

Comment (PatKamin) [reply RELEASE_STEPS.md:17]: Maybe rephrase it so that the information that UMF will be released once for each oneAPI release is included?

**Review by bratpiorka — APPROVED** [review]

File: RELEASE_STEPS.md (hunk 4 above)
Comment (lukaszstolarczuk) [reply RELEASE_STEPS.md]: Done.

**Review by kswiecicki — APPROVED** [review]

**Review by szadam — APPROVED** [review]

**Review by PatKamin — APPROVED** [review]
